app/src/ui/dispatcher/dispatcher.ts

```diff
@@ -1400,99 +1404,106 @@ export class Dispatcher {
     }
   }
 
-  private async getForkAndUpstreamRepos(url: string) {
+  /**
+   * Find an existing repository that can be used for checking out
+   * the passed pull request.
+   *
+   * @param pullRequest the pull request object received from the API.
+   */
+  private getRepositoryFromPullRequest(
+    pullRequest: IAPIPullRequest
+  ): Repository | null {
     const state = this.appStore.getState()
     const repositories = state.repositories
+    const headUrl = pullRequest.head.repo && pullRequest.head.repo.clone_url
+    const baseUrl = pullRequest.base.repo && pullRequest.base.repo.clone_url
+
+    if (headUrl === null || baseUrl === null) {
+      return null
+    }
+
+    for (const repository of repositories) {
+      if (this.doesRepositoryMatchUrl(repository, headUrl)) {
+        return repository
+      }
+    }
```

Comment:
I wanted to use `repositories.find()` here but TypeScript's `Array.prototype.find` is not capable to infer that the found element can only be of type `Repository | null`.